src/braket/tasks/gate_model_quantum_task_result.py

```diff
@@ -332,6 +349,7 @@ def _calculate_result_types(
                 value = GateModelQuantumTaskResult._probability_from_measurements(
                     measurements, measured_qubits, targets
                 )
+                casted_result_type = Probability(targets=targets)
```

Comment:
What is this casting for?

Reply:
result_type is a dict and ResultTypeValue.type expects an ir Result